pkg/oc/cli/cmd/rollout/retry.go

```diff
@@ -56,77 +62,98 @@ var (
 `)
 )
 
-func NewCmdRolloutRetry(fullName string, f kcmdutil.Factory, out io.Writer) *cobra.Command {
-	opts := &RetryOptions{}
+func NewRolloutRetryOptions(streams genericclioptions.IOStreams) *RetryOptions {
+	return &RetryOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("already retried").WithTypeSetter(ocscheme.PrintingInternalScheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewCmdRolloutRetry(fullName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewRolloutRetryOptions(streams)
 	cmd := &cobra.Command{
 		Use:     "retry (TYPE NAME | TYPE/NAME) [flags]",
 		Long:    rolloutRetryLong,
 		Example: fmt.Sprintf(rolloutRetryExample, fullName),
 		Short:   "Retry the latest failed rollout",
 		Run: func(cmd *cobra.Command, args []string) {
-			kcmdutil.CheckErr(opts.Complete(f, cmd, out, args))
-			kcmdutil.CheckErr(opts.Run())
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
 		},
 	}
+
+	o.PrintFlags.AddFlags(cmd)
+
 	usage := "Filename, directory, or URL to a file identifying the resource to get from a server."
-	kcmdutil.AddFilenameOptionFlags(cmd, &opts.FilenameOptions, usage)
+	kcmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, usage)
 	return cmd
 }
 
-func (o *RetryOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, out io.Writer, args []string) error {
+func (o *RetryOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
 	var err error
 	if len(args) == 0 && len(o.FilenameOptions.Filenames) == 0 {
 		return kcmdutil.UsageErrorf(cmd, cmd.Use)
 	}
 
-	o.Typer = legacyscheme.Scheme
+	o.Resources = args
+
 	o.Mapper, err = f.ToRESTMapper()
 	if err != nil {
 		return err
 	}
 	o.Encoder = kcmdutil.InternalVersionJSONEncoder()
-	o.Out = out
 
-	cmdNamespace, enforceNamespace, err := f.ToRawKubeConfigLoader().Namespace()
+	o.Namespace, o.ExplicitNamespace, err = f.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
 		return err
 	}
 
-	clientConfig, err := f.ToRESTConfig()
+	o.Clientset, err = f.ClientSet()
 	if err != nil {
 		return err
 	}
-	o.Clientset, err = kclientset.NewForConfig(clientConfig)
+
+	o.Printer = func(msg string) (printers.ResourcePrinter, error) {
+		if err := o.PrintFlags.Complete(msg); err != nil {
```

Comment:
Hmmm... with `NamePrinter` this will require a template that is passed to `Sprintf` call, double check if this change works the way you plan.